src/lazy/text/encoded_value.rs

```diff
@@ -60,6 +62,8 @@ pub(crate) struct EncodedTextValue {
     // value is stored. For others (e.g. a timestamp), the various components of the value are
     // recognized during matching and partial information like subfield offsets can be stored here.
     matched_value: MatchedValue,
+
+    field_name_syntax: Option<MatchedSymbol>,
```

Comment:
1. Will do.
2. I'm not sure I understand the suggestion, but I can explain the approach. Both the binary version and the text version model their lazy values as a `(field_id?, annotations?, value)` tuple. That is, the field is modeled as being part of the value. (This language is loose; the data structures and approach to parsing couple the field with the value but I recognize that "the field is part of the value" doesn't feel quite right from an Ion data model perspective.) If I did introduce a `LazyRawTextSymbolToken` type, it would simply wrap a `MatchedSymbol`. Let me know if you'd like to discuss this further, I think an offline discussion would probably be helpful.
3. I think it's possible, yeah. Might even be a small performance win by shrinking the size of values that aren't in structs. I've opened #631 to track investigating this.

Reply:
Opened #632 to track investigating an optimized representation for annotations offset info.